src/core/lib/iomgr/tcp_server_posix.cc

```diff
@@ -460,6 +549,17 @@ static grpc_error_handle tcp_server_add_port(grpc_tcp_server* s,
       }
     }
   }
+
+#ifdef HAVE_LIBSYSTEMD
```

Comment:
instead of this `ifdef`, you can define stubs that just return after the previous `ifdef`, in an `#else`, so that the functions are always defined